components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -234,6 +238,39 @@ public Application updateApplication(ApplicationUpdateRequest updateRequest, Str
                 String backChannelLogoutUri = validateBackchannelLogoutURI(updateRequest.getBackchannelLogoutUri());
                 appDTO.setBackChannelLogoutUrl(backChannelLogoutUri);
             }
+            if (CollectionUtils.isNotEmpty(updateRequest.getAud())) {
+                appDTO.setAudiences(updateRequest.getAud().toArray(new String[0]));
+            }
+
+            if (validateTokenEndpointAuthMethod(updateRequest.getTokenEndpointAuthMethod())) {
+                appDTO.setTokenEndpointAuthMethod(updateRequest.getTokenEndpointAuthMethod());
+            }
+
+            if (updateRequest.getIdTokenEncryptionAlgorithm() != null ||
```

Comment:
shall we use `StringUtils.isNotBlank` here

Reply:
In this implementation if someone send `id_token_encrypted_response_alg` as a blank value (" ") in the update request, IS will not update the value and throw an error. But if we use `StringUtils.isNotBlank` IS will not update the value but will return a 200 response. Is that ok ?

Reply:
because `StringUtils.isBlank(" ") = true`

Reply:
thanks for the explanation.
adding for reference: `id_token_encrypted_response_alg = " "` will be validated at https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java#L1392

Reply:
@mefarazath the ideal way to handle `id_token_encrypted_response_alg = " "` would be to throw an error instead of ignoring it right (just wanted to confirm the recommended approach)?

Reply:
Yes. Throwing a Bad request error would be correct way since the user sends an invalid value